jwst/assign_wcs/nirspec.py

```diff
@@ -338,16 +338,17 @@ def get_open_fixed_slits(input_model):
     return slits
 
 
-def get_msa_metadata(input_model):
+def get_msa_metadata(input_model, reference_files):
     """
     Get the MSA metadata file (MSAMTFL) and the msa metadata id (MSAMETID).
 
     """
-    msa_config = input_model.meta.instrument.msa_metadata_file
-    if msa_config is None:
-        message = "MSA metadata file is not available (keyword MSAMETFL)."
+    try:
+        msa_config = reference_files['msametafile']
+    except KeyError:
```

Comment:
Thinking about this some more, here the fall back should be getting the file name from the model and assuming it is run from the current directory (essentially duplicating the current behaviour). If this fails then raise the error. The reason for this is that there are (limited) cases when this function is called directly from code outside assign_wcs and the call may bypass assign_wcs_step. 

Another option would be to somehow pass `input_dir` downstream.

Reply:
Makes sense.  I will update the PR.

I avoided passing `Step.input_dir` downstream, as that would have changed several function calls.